generate.py

```diff
@@ -57,29 +59,47 @@ def main():
         residual_channels=wavenet_params['residual_channels'],
         dilation_channels=wavenet_params['dilation_channels'],
         quantization_channels=wavenet_params['quantization_channels'],
-        use_biases=wavenet_params['use_biases'])
+        use_biases=wavenet_params['use_biases'],
+        fast_generation=args.fast_generation)
 
     samples = tf.placeholder(tf.int32)
 
     next_sample = net.predict_proba(samples)
-
-    saver = tf.train.Saver()
+    
+    if args.fast_generation:
+        sess.run(tf.initialize_all_variables())
+        sess.run(net.init_ops)
+
+    variables_to_restore = {var.name[:-2]: var for var in tf.all_variables() if 'Variable' in var.name}
+    saver = tf.train.Saver(variables_to_restore)
+    
     print('Restoring model from {}'.format(args.checkpoint))
     saver.restore(sess, args.checkpoint)
 
     decode = net.decode(samples)
 
-    quantization_steps = wavenet_params['quantization_steps']
-    waveform = np.random.randint(quantization_steps, size=(1,)).tolist()
+    quantization_channels = wavenet_params['quantization_channels']
+    waveform = np.random.randint(quantization_channels, size=(1,)).tolist()
     for step in range(args.samples):
-        if len(waveform) > args.window:
-            window = waveform[-args.window:]
+        if args.fast_generation:
```

Comment:
I suggest simplify this line by extracting fast-generation-specific operation, like:

```python
if args.fast_generation:
    window = waveform[-1]
    outputs = [next_sample, net.push_ops]
else:
    window = waveform[-args.window:]
    outputs = [next_sample]
prediction = sess.run(outputs, feed_dict={samples: window})[0]
```